lib/mailman/route/conditions.rb

```diff
@@ -4,8 +4,14 @@ class Route
     # Matches against the To addresses of a message.
     class ToCondition < Condition
       def match(message)
-        if !message.to.nil?
+        # this is done for cases where the `to` field is missing but does
+        # have a `Delivered-To` - so we pick from that instead
+        unless message.to.nil? && message.header['Delivered-To'].nil?
           messageto = message.to.is_a?(Array) ? message.to : [message.to]
+          messageto_from_headers = [message.header['Delivered-To']].flatten
+                                                                   .compact
+                                                                   .map(&:to_s)
+          messageto.concat(messageto_from_headers)
```

Comment:
```suggestion
          messageto.concat(message_delivered_to)
```

Reply:
sticking to the current convention since it is extracting the messageto attribute from the headers.